cgsmiles/resolve.py

```diff
@@ -254,12 +260,18 @@ def resolve(self):
         else:
             all_atom = False
 
-        # empty the fragment dict just as a precaution
-        self.fragment_dict = {}
+        # get the next set of fragments
+        if self.fragment_strs:
+            fragment_str = next(self.fragment_strs)
```

Comment:
```suggestion
        # get the next set of fragments
        if self.fragment_strs:
            fragment_str = self.fragment_strs[self.resolution_counter - 1]  # -1 because the counter has already been incremented #FIXME
```